components/post/PostDetails.tsx

```diff
@@ -0,0 +1,158 @@
+'use client';
+
+import PostItemBadge from './PostItemBadge';
+
+import ChecklistItem from '../shared/ChecklistItem';
+import ComponentMenu from '../shared/ComponentMenu';
+import LearningResourceItem from '../shared/LearningResourceItem';
+import SyntaxHighlightAndCopy from '../shared/SyntaxHighlightAndCopy';
+import TagItem from '../shared/TagItem';
+import { useToast } from '../ui/use-toast';
+
+import { format } from 'date-fns';
+import parse, {
+  Element,
+  HTMLReactParserOptions,
+  Text,
+} from 'html-react-parser';
+import Image from 'next/image';
+import { useRouter } from 'next/navigation';
+
+import { deletePost } from '@/lib/actions/post-actions';
+import type { IPost } from '@/types/post';
+import { EPostType } from '@/types/post-types';
+
+// ----------------------------------------------------------------
+
+interface IPostDetailsProps {
+  post: IPost;
+}
+
+const parserOptions: HTMLReactParserOptions = {
+  replace: (domNode) => {
+    // if user adds code to the Rich Text Editor, format it with SyntaxHighlighter
+    if (domNode instanceof Element && domNode.name === 'pre') {
+      const codeTag = domNode.children.find(
+        (node) => node instanceof Element && node.name === 'code'
+      );
+
+      if (codeTag) {
+        const codeData = (
+          (codeTag as Element).children.find(
+            (node) => node instanceof Text && node.data
+          ) as Text | undefined
+        )?.data;
+
+        if (!codeData) return domNode;
+
+        return <SyntaxHighlightAndCopy code={codeData} />;
+      }
+    }
+
+    if (domNode instanceof Element && domNode.name === 'a') {
+      domNode.attribs.class = 'text-primary-500 underline';
+      return domNode;
+    }
+  },
+};
+
+const PostDetails: React.FC<IPostDetailsProps> = ({ post }) => {
+  const { toast } = useToast();
+  const router = useRouter();
+
+  const {
+    title,
+    type,
+    description,
+    tags,
+    codeExample,
+    content,
+    checklist,
+    learningResources,
+    createdAt,
+    _id,
+  } = post;
+
+  const checklistTitle =
+    type === EPostType.KNOWLEDGE ? 'Key Takeaways' : 'Task Checklist';
+
+  const handleDeletePost = async () => {
+    try {
+      const response = await deletePost(_id);
+      if (response?.ok && response.status === 200) {
+        toast({ variant: 'success', title: 'Post deleted successfully!' });
+        router.push('/');
+      }
```

Comment:
I have handled the case now but i am wondering if it could be handled like in the case above ^^^ with update/create posts. Where i throw error in server actions and just catch and display them on the FE?